client/client.go

```diff
@@ -180,11 +185,11 @@ func (cli *Client) Close() error {
 // It appends the query parameters to the path if they are not empty.
 func (cli *Client) getAPIPath(ctx context.Context, p string, query url.Values) string {
 	var apiPath string
-	if cli.negotiateVersion && !cli.negotiated {
+	if cli.negotiateVersion && !cli.isVersionNegotiated() {
 		cli.NegotiateAPIVersion(ctx)
 	}
-	if cli.version != "" {
-		v := strings.TrimPrefix(cli.version, "v")
+	if v := cli.ClientVersion(); v != "" {
+		v = strings.TrimPrefix(v, "v")
```

Comment:
If `cli.version` is now only set through `setVersion`, perhaps we should move the `TrimPrefix` there

Reply:
It makes sense, but I'm concerned about making this change, because:
1. It causes test failures:
```
=== FAIL: client TestGetAPIPath (0.00s)
    client_test.go:133: assertion failed: /vv1.22/containers/json (actual string) != /v1.22/containers/json (testcase.expected string)
    client_test.go:133: assertion failed: /vv1.22/containers/json (actual string) != /v1.22/containers/json (testcase.expected string)
    client_test.go:133: assertion failed: /vv1.22/containers/json?s=c (actual string) != /v1.22/containers/json?s=c (testcase.expected string)
    client_test.go:133: assertion failed: /vv1.22/networks/kiwl$%25%5E (actual string) != /v1.22/networks/kiwl$%25%5E (testcase.expected string)
```
2. It implies that every other version usage expects a version without `v` prefix, and I think this is a dangerous assumption: users can set their versions using `WithVersion("v1.xx")` and expect `v1.xx` in return from `ClientVersion`. Trimming `v` prefix in `setVersion` will change the behavior.